.gitlab-ci.yml

```diff
@@ -157,6 +158,20 @@ before_script:
     variables:
       - $RELEASE_VERSION_7 == ""
 
+# Fail if we're running a pipeline on a non-triggered tag build
+# NOTE: All jobs with 'needs' dependencies should also 'need' this to workaround a Gitlab issue: https://gitlab.com/gitlab-org/gitlab/issues/31526
```

Comment:
This means that the pipelines will break the day Gitlab is updated with the fix for this issue, but that's fine since we'll then be able to remove this hack